core/org.wso2.carbon.user.core/src/test/java/org/wso2/carbon/user/core/jdbc/AdvancedJDBCRealmTest.java

```diff
@@ -313,16 +313,6 @@ public void doUserRoleStuff() throws Exception {
            // Renaming Role
            admin.updateRoleName("role4", "role5");
 
-        try {
-            // updating to invalid role name
-            admin.updateRoleName("role5", "role@12#$");
-            fail("Able to rename role with invalid characters");
```

Comment:
Can we see how we can test for both shared vs non-shared instances? This is configured via a property ("SharedGroupEnabled") realmConfig property in user-mgt.xml? Removing an existing test does not seem correct. :)

Reply:
This check is not necessary for non-shared roles. Hence changed the test method to fail only if shared roles are enabled